molo/yourwords/wagtail_hooks.py

```diff
@@ -0,0 +1,89 @@
+from django.http import HttpResponse
+from import_export import resources
+from molo.yourwords.admin import YourWordsCompetitionAdmin
+from molo.yourwords.models import YourWordsCompetitionEntry, \
+    YourWordsCompetition
+from wagtailmodeladmin.options import ModelAdmin, wagtailmodeladmin_register, \
+    ModelAdminGroup
+from wagtailmodeladmin.views import IndexView
+
+
+class YourWordsEntriesResource(resources.ModelResource):
+    class Meta:
+        model = YourWordsCompetitionEntry
+        exclude = ('id', '_convert', 'article_page')
+
+
+class ModelAdminTemplate(IndexView):
+    def post(self, request, *args, **kwargs):
+        list_filter = request.GET.get('competition__slug')
+
+        dataset = YourWordsEntriesResource().export(
+            YourWordsCompetitionEntry.objects.filter(
+                competition__slug=list_filter
+            ) if list_filter is not None else None)
+
+        response = HttpResponse(dataset.csv, content_type="csv")
+        response['Content-Disposition'] = \
+            'attachment; filename=yourwords_entries.csv'
+        return response
+
+    def get_template_names(self):
+        return 'admin/yourwords/model_admin_template.html'
+
+
+class YourWordsEntriesModelAdmin(ModelAdmin):
+    model = YourWordsCompetitionEntry
+    menu_label = 'Entries'
+    menu_icon = 'edit'
+    index_view_class = ModelAdminTemplate
+    add_to_settings_menu = False
+    list_display = ['story_name', 'user', 'hide_real_name',
+                    'submission_date', 'is_read', 'is_shortlisted',
+                    'is_winner', '_convert']
+
+    list_filter = ['competition__slug']
```

Comment:
Can we filter here with ['submission_date', 'is_read', 'is_shortlisted', 'is_winner'] instead